include/rapidcheck/Gen.hpp

```diff
@@ -55,7 +68,12 @@ class Gen<T>::GenImpl : public IGenImpl {
 template <typename T>
 template <typename Impl, typename>
 Gen<T>::Gen(Impl &&impl)
-    : m_impl(new GenImpl<Decay<Impl>>(std::forward<Impl>(impl))) {}
+    : m_impl(new GenImpl<Decay<Impl>>(std::forward<Impl>(impl))) {
+  static_assert(MakesShrinkable<T, Decay<Impl>>::value,
+                "Generator implementation must have a method Shrinkable<T> operator()(const Random &, int) const");
+  static_assert(std::is_copy_constructible<Decay<Impl>>::value,
```

Comment:
It shouldn't actually need to be copyable, have you tried not having a copy constructor? I know the doc comment says it does but I think that's incorrect.